lib/puppet/provider/package/yum.rb

```diff
@@ -22,7 +22,8 @@
       end
   end
 
-defaultfor :osfamily => :redhat, :operatingsystemmajrelease => (4..7).to_a
+# Amazon Linux reports operatingsystemmajrelease as "2018"; Amazon Linux 2 reports "2"
+defaultfor :osfamily => :redhat, :operatingsystemmajrelease => [2, 4, 5, 6, 7, 8, 2018]
```

Comment:
It looks like `8` was added to this list.  RHEL 8 uses the dnf provider, so it shouldn't be in this array.

Reply:
Removed. Thanks!